docs/wave.rst

```diff
@@ -52,55 +53,94 @@ First of all, we will set up the initial condition for damping::
                    time_order=time_order, space_order=spc_order, save=True,
                    dtype=damp.dtype)
 
 TimeData is a devito data object used to store and manage time-varying data
 
-We initialise our grid to be of size :samp:`model.get_shape_comp()` which is a 3-D tuple.
-:obj:`time_dim` represents the size of the time dimension that dictates the leading dimension of the data buffer.
-:obj:`time_order` and :obj:`space_order` represent the discretization order for time and space respectively.
+We initialize our grid to be of size :samp:`model.get_shape_comp()` which is a 3-D tuple.
+:obj:`time_dim` represents the size of the time dimension that dictates
+the leading dimension of the data buffer.
+:obj:`time_order` and :obj:`space_order` represent the discretization order
+for time and space respectively.
 
 The next step is to generate the stencil to be solved by a :obj:`devito.operator.Operator`
 
 The stencil is generated according to Devito conventions. It uses a sympy
 expression to represent the acoustic wave equation. Devito makes it easy to
 represent the equation in a finite-difference form by providing properties :obj:`dt2`, :obj:`laplace`, :obj:`dt`.
-We then generate the stencil by solving eqn for u.forward, a symbol for the time-forward state of the function.
+We then generate the stencil by solving eqn for :obj:`u.forward = u(t+dt,x,y,z)`,
+a symbol for the time-forward state of the function.
 ::
 
   from devito import Operator
   from sympy import Eq, solve, symbols
   eqn = m*u.dt2-u.laplace+damp*u.dt
   stencil = solve(eqn, u.forward)[0]
 
-We plug the stencil in an Operator, as shown, and define the values the spacing between cells :obj:`h` and the
-temporal spacing :obj:`s`.::
+We plug the stencil in an Operator, as shown, and define the values of the spacing
+between cells :obj:`h` and the temporal spaces :obj:`s`. ::
 
   s, h = symbols('s h')
   subs = {s: model.get_critical_dt(), h: model.get_spacing()}
-  super(AdjointOperator, self).__init__(nt, m.shape,
-                                        stencils=Eq(v.backward, stencil),
-                                        substitutions=subs,
-                                        spc_border=spc_order/2,
-                                        time_order=time_order,
-                                        forward=False,
-                                        dtype=m.dtype,
-                                        **kwargs)
-
+  super(ForwardOperator, self).__init__(nt, m.shape,
+                                              stencils=Eq(u.forward, stencil),
+                                              subs=subs,
+                                              spc_border=spc_order/2,
+                                              time_order=time_order,
+                                              forward=True,
+                                              dtype=m.dtype,
+                                              **kwargs)
+
```

Comment:
Maybe a few details on
 `spc_border` : ghost points at the edge of the domain to be able touse the same stencil everywhere
`forward` : propagation time direction (True, forward in time, False, backward in time)
`nt` : number of time steps